core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -46,6 +46,10 @@ oppia.factory('UrlService', ['$window', function($window) {
     getPathname: function() {
       return this.getCurrentLocation().pathname;
     },
+    getTopicIdFromUrl: function() {
+      var pathname = this.getPathname();
+      return pathname.split('/')[2].substring(0, 12);
```

Comment:
Add unit tests for this.

Also, why wouldn't it be the whole of the second component? Better to take the second component, check whether it has length 12, throw an error if not ("The URL xxx is not a valid topic URL"), and otherwise return it.

Reply:
I thought of that, but using tabs, I saw hashes in front of the id. So, to ignore that, used the first 12 characters.

Reply:
Done, added tests.

Reply:
I am merging now, will change in this PR (https://github.com/oppia/oppia/pull/5119) if something else can be done.

Reply:
Read up on the structure of URLs -- the hash stuff is a different part, and can be separated out (window.location.hash I think). So, extract only the main part of the URL (without the hash), and take the relevant component of that (validating as described above). This needs to be fixed in a follow-up PR.